src/test/buffer_test.cpp

```diff
@@ -0,0 +1,96 @@
+#include "buffer.h"
+
+#include <catch2/catch.hpp>
+#include <string_view>
+
+namespace {
+
+// This could become the beginnings of a C++ wrapper over a C implementation...
+// And then eventually a reimplementation in C++ (which maybe a C shim if needed).
+class Buffer {
```

Comment:
OK so this is just a "demo" class to illustrate testability.

Reply:
Ah I see, it wraps the older buffer code.

Reply:
Totally. It's there to show how it could be, while using the exact same code.